pennylane/shadows/classical_shadow.py

```diff
@@ -316,22 +317,139 @@ def qnode(x):
+                Thus, ``k=1`` is default and advised.
+            base (float): Base to the logarithm used for the entropies.
+            atol (float): Absolute tolerance for Eigenvalues close to 0 that are taken into account.
+
+        Returns:
+            float: Entropy of the chosen subsystem.
+
+        **Example**
+
+        For the maximally entangled state of ``n`` qubits, the reduced state has two constant eigenvalues :math:`\frac{1}{2}`. For constant distributions, all Renyi entropies are
+        equivalent:
+
+        .. code-block:: python3
+
+            wires=4
+
+            @qml.qnode(dev)
+            def max_entangled_circuit():
+                qml.Hadamard(wires=0)
+                for i in range(1, wires):
+                    qml.CNOT(wires=[0, i])
+                return qml.classical_shadow(wires=range(wires))
+
+            bits, recipes = max_entangled_circuit()
+            shadow = ClassicalShadow(bits, recipes)
+
+            entropies = [shadow.entropy(wires=[0], alpha=alpha, atol=1e-2) for alpha in [1., 2., 3.]]
+
+        >>> np.isclose(entropies, entropies[0], atol=1e-2)
+        [ True,  True,  True]
+
+        For non-uniform reduced states that is not the case anymore and the entropy differs for each order ``alpha``:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def qnode(x):
+                for i in range(n_wires):
+                    qml.RY(x[i], wires=i)
+
+                for i in range(n_wires-1):
+                    qml.CNOT((i,i+1))
+
+                return classical_shadow(wires=dev.wires)
+
+            bitstrings, recipes = qnode(x)
+            shadow = ClassicalShadow(bitstrings, recipes)
+
+            entropies = [shadow.entropy(wires=wires, alpha=alpha, atol=1e-10) for alpha in [1., 2., 3.]]
+
+        >>> print(entropies)
+        [0.6727522114759635, 0.6252047673044356, 0.5996592216299593]
+
+        """
+        global_snapshots = self.global_snapshots(wires=wires, snapshots=snapshots)
+        rdm = median_of_means(global_snapshots, k, axis=0)
+
+        # Allow for different log base
+        if base:
+            div = np.log(base)
+        else:
+            div = 1
+
+        if alpha == 2:
+            # special case of purity
+            res = -qml.math.log(qml.math.trace(rdm @ rdm))
+            res = qml.math.cast(res, dtype="float64")
+            return res / div
+
+        # Else
+        # Compute Eigenvalues and choose only those >>0
+        evs = qml.math.eigvalsh(rdm)
+        mask0 = qml.math.logical_not(qml.math.isclose(evs, 0, atol=atol))
+        mask1 = qml.math.where(evs > 0, True, False)
+        mask = qml.math.logical_and(mask0, mask1)
+        # Renormalize
+        evs_nonzero = qml.math.gather(evs, mask)
+        evs_nonzero = evs_nonzero / qml.math.sum(evs_nonzero)
```

Comment:
Maybe add an inline comment on why we do this normalization. (If I'm not mistaken, it's because the reconstructed density matrix doesn't have trace precisely 1?)

Just out of curiosity, have you checked the results if we don't do this normalization?